exporters/prometheus/src/prometheus_exporter_utils.cc

```diff
@@ -0,0 +1,467 @@
+ */
+template <typename T>
+void PrometheusExporterUtils::SetData(std::vector<T> values,
+                                      const std::vector<double> &boundaries,
+                                      const std::vector<int> &counts,
+                                      const std::string &labels,
+                                      std::chrono::nanoseconds time,
+                                      prometheus_client::MetricFamily *metric_family)
+{
+  metric_family->metric.emplace_back();
+  prometheus_client::ClientMetric &metric = metric_family->metric.back();
+  SetMetricBasic(metric, time, labels);
+  SetValue(values, boundaries, counts, &metric);
+}
+
+/**
+ * Set metric data for:
+ * MinMaxSumCount => Prometheus Gauge
+ * Use Average (sum / count) as the gauge metric
+ */
+void PrometheusExporterUtils::SetData(double value,
+                                      const std::string &labels,
+                                      std::chrono::nanoseconds time,
+                                      prometheus_client::MetricFamily *metric_family)
+{
+  metric_family->metric.emplace_back();
+  prometheus_client::ClientMetric &metric = metric_family->metric.back();
+  SetMetricBasic(metric, time, labels);
+  SetValue(value, &metric);
+}
+
+/**
+ * Set metric data for:
+ * Exact => Prometheus Summary
+ * Sketch => Prometheus Summary
+ */
+template <typename T>
+void PrometheusExporterUtils::SetData(std::vector<T> values,
+                                      metric_sdk::AggregatorKind kind,
+                                      const std::vector<T> &quantiles,
+                                      const std::string &labels,
+                                      std::chrono::nanoseconds time,
+                                      prometheus_client::MetricFamily *metric_family,
+                                      bool do_quantile,
+                                      std::vector<double> quantile_points)
+{
+  metric_family->metric.emplace_back();
+  prometheus_client::ClientMetric &metric = metric_family->metric.back();
+  SetMetricBasic(metric, time, labels);
+  SetValue(values, kind, quantiles, &metric, do_quantile, quantile_points);
+}
+
+/**
+ * Set time and labels to metric data
+ */
+void PrometheusExporterUtils::SetMetricBasic(prometheus_client::ClientMetric &metric,
+                                             std::chrono::nanoseconds time,
+                                             const std::string &labels)
+{
+  metric.timestamp_ms = time.count() / 1000000;
+
+  auto label_pairs = ParseLabel(labels);
+  if (!label_pairs.empty())
+  {
+    metric.label.resize(label_pairs.size());
+    for (int i = 0; i < label_pairs.size(); ++i)
+    {
+      auto origin_name = label_pairs[i].first;
+      auto sanitized   = SanitizeNames(origin_name);
+      if (origin_name != sanitized)
+      {
+        std::cout << "Sanitized label name \"" << origin_name << "\" to \"" << sanitized << "\"\n";
```

Comment:
Same here, need to understand if we intend to keep the cout lines.

Reply:
removed.